### PR TITLE
Add test for OS updates being staged within 5-10m.

### PR DESCRIPTION
**Pull request by dgoodwin**
Recent test failure investigation led to a possibility of slow disk
speeds doing OS content extraction. Add a test to ensure that we move
from OSUpdateStarted to OSUpdateStaged within 5-10 minutes, which
primarily just encompasses extraction of the content.

If any node is over 10 minutes we will fail the test, if some are over 5
we'll flake instead.

### CODE REVIEW

File: pkg/synthetictests/operators.go [hunk 1]
```diff
@@ -73,6 +73,77 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
 	return ret
 }
 
+func testOperatorOSUpdateStaged(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	testName := "[bz-Machine Config Operator] Nodes should reach OSUpdateStaged in a timely fashion"
+	success := &junitapi.JUnitTestCase{Name: testName}
+
+	type startedStaged struct {
+		OSUpdateStarted time.Time
```
Comment (deads2k) [inline pkg/synthetictests/operators.go:81]: give me some comments about what this state means

File: pkg/synthetictests/operators.go [hunk 2]
```diff
@@ -73,6 +73,77 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
...
+		OSUpdateStaged  time.Time
```
Comment (deads2k) [inline pkg/synthetictests/operators.go:82]: give me some comments about what this state means

**Comment by deads2k** [comment]
/approve

a request for some comments and lgtm

File: pkg/synthetictests/operators.go [hunk 3]
```diff
@@ -73,6 +73,82 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
...
+
+	// Iterate the data we assembled looking for any nodes with an excessive time between Started/Staged, or those
+	// missing a Staged
+	slowStageMessages := []string{}
+	var failTest bool // set true if we see anything over 10 minutes, our failure threshold
+	for node, ss := range nodeOSUpdateTimes {
+		if ss.OSUpdateStaged.IsZero() {
+			slowStageMessages = append(slowStageMessages, fmt.Sprintf("%s OSUpdateStarted at %s, did not make it to OSUpdateStaged", node, ss.OSUpdateStarted))
+			failTest = true
+		} else if ss.OSUpdateStaged.Sub(ss.OSUpdateStarted) > 5*time.Minute {
```
Comment (wking) [inline pkg/synthetictests/operators.go:122]: possible worth an additional case here for `ss.OSUpdateStarted.IsZero()` for "best-effort event submission must have dropped the started event, so I dunno how long this node too to get from there to staged".  I guess I'd expect that to be a flake, but should be rare enough that if we decide to fail on it the number of runs that fail as a result should be small.

**Comment by dgoodwin** [comment]
@wking has pointed out that we may have suites that do multi-stage updates, meaning we move through these conditions several times. I'm also not sure this currently properly handles install time events. Will revisit asap tomorrow.

File: pkg/synthetictests/operators.go [hunk 4]
```diff
@@ -73,6 +73,82 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
...
+
+			// Separate check, did we see anything over 10 minutes, in which case we want to fail the test
+			if ss.OSUpdateStaged.Sub(ss.OSUpdateStarted) > 10*time.Minute {
+				failTest = true
+			}
+		}
+
+	}
+	if len(slowStageMessages) > 0 {
+		output := fmt.Sprintf("%d nodes took over 5 minutes to stage OSUpdate:\n\n%s",
```
Comment (wking) [inline pkg/synthetictests/operators.go:134]: might be a local variable for `5*time.Minute` that we can format in here, so folks don't bump the threshold but forget to bump this output string.

**Comment by deads2k** [comment]
> @wking has pointed out that we may have suites that do multi-stage updates, meaning we move through these conditions several times. I'm also not sure this currently properly handles install time events. Will revisit asap tomorrow.

followup is acceptable to me.

**Comment by deads2k** [comment]
cleaning up CI signal on tests

**Comment by dgoodwin** [comment]
/hold 

Something's wrong

**Comment by dgoodwin** [comment]
Fixed a pointer bug causing zero start times and empty staged times, added vars for the flake/fail thresholds, added the check for edge case on zero start time (which will fail the test for now), formatted timestamps better for printing, and *may* have handled test suites with multiple update cycles.
